raft/src/main/scala/zio/raft/Raft.scala

```diff
@@ -576,17 +575,20 @@ class Raft[A <: Command](
         .stream(state.lastApplied.plusOne, state.commitIndex)
         .runFoldZIO(state)((state, logEntry) =>
           for
-            response <- stateMachine.modify(_.apply(logEntry.command))
+            appState <- appStateRef.get
+            (newState, response) <- stateMachine.apply(logEntry.command).toZIOWithState(appState).orDieWith(e => new RuntimeException(s"Failed to apply command ${logEntry.command} to state machine: $e")) // TODO (Eran): Remove orDie, we should propagate the error instead
```

Comment:
instead of orDie change it to `.either`, and fail the promise later on. Error is a valid response...

Reply:
yea I left a TODO, it was just so it will compile :)